https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/4df753d0d7af7e68ab16ee90b94c4ddd44506074

From 4df753d0d7af7e68ab16ee90b94c4ddd44506074 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Mon, 26 Jun 2023 16:11:33 +0200
Subject: [PATCH] context: avoid segfault when no fallback driver

In some cases, there might not be a fallback driver. Handle this without
segfaulting.
---
 src/pipewire/context.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pipewire/context.c b/src/pipewire/context.c
index daf13c16b..5fd0cdec9 100644
--- a/src/pipewire/context.c
+++ b/src/pipewire/context.c
@@ -1327,11 +1327,11 @@ again:
 			if ((t->want_driver && t->active && t->runnable) ||
 			    t->always_process) {
 				driver = target;
-				driver->runnable = true;
 				break;
 			}
 		}
 		if (driver != NULL) {
+			driver->runnable = true;
 			/* driver needed for this group */
 			move_to_driver(context, &collect, driver);
 		} else {
-- 
GitLab
